app/models/project.rb

```diff
@@ -100,8 +101,22 @@ def with_lock(output: StringIO.new, holder:, error_callback: nil, timeout: 10.mi
     MultiLock.lock(id, holder, timeout: timeout, failed_to_lock: callback, &block)
   end
 
+  def last_deploy_by_group
+    releases = deploys_by_group
+    releases.map { |k, deploys| [ k, deploys.sort_by { |deploy| deploy.updated_at }.reverse.first ] }.to_h
```

Comment:
can use proc shorthand for sort and reverse.first == last
`releases.map { |group_id, deploys| [ group_id, deploys.sort_by(&:updated_at).last ] }.to_h`